src/web/mod.rs

```diff
@@ -84,6 +88,50 @@ async fn websocket_connection(socket: WebSocket, state: AppState) {
     send_task.await.unwrap();
 }
 
+#[derive(Deserialize)]
+struct FrequencyQuery {
+    frequency: Option<u8>,
+}
+
+async fn stats_websocket_handler(
+    ws: WebSocketUpgrade,
+    State(state): State<AppState>,
+    Query(freq_query): Query<FrequencyQuery>,
+) -> Response {
+    ws.on_upgrade(
+        |socket| async move { stats_websocket_connection(socket, state, freq_query).await },
+    )
+}
+
+async fn stats_websocket_connection(
+    socket: WebSocket,
+    state: AppState,
+    freq_query: FrequencyQuery,
+) {
+    let frequency = freq_query.frequency.unwrap_or(1);
+    let identifier = Uuid::new_v4();
+    debug!("WS client connected with ID: {identifier}");
+
+    let (mut sender, mut receiver) = socket.split();
+
+    let interval_duration = tokio::time::Duration::from_secs_f32(1.0 / frequency as f32);
+    let periodic_task = tokio::spawn(async move {
+        let mut interval = tokio::time::interval(interval_duration);
+        loop {
+            interval.tick().await;
+            let json = serde_json::to_string(&hub::hub_messages_stats().await.unwrap()).unwrap();
```

Comment:
maybe something like:
```suggestion
            let hub_message_stats = match hub::hub_messages_stats().await {
                Ok(hub_message_stats) => hub_message_stats,
                Err(error) => {
                    warn!("Failed getting hub message stats: {error:?}");
                    continue;
                }
            }
            let json = match serde_json::to_string(&hub_message_stats) {
                Ok(json) => json,
                Err(error) => {
                    warn!("Failed to create json from Hub Message Stats: {error:?});
                    continue;
                }
            };
```